fawltydeps/packages.py

```diff
@@ -181,21 +192,26 @@ def lookup_packages(self, package_names: Set[str]) -> Dict[str, Package]:
 class LocalPackageResolver(BasePackageResolver):
     """Lookup imports exposed by packages installed in a Python environment."""
 
-    def __init__(self, pyenv_path: Optional[Path] = None) -> None:
+    def __init__(self, pyenv_paths: AbstractSet[Path] = frozenset()) -> None:
         """Lookup packages installed in the given virtualenv.
 
-        Default to the current python environment if `pyenv_path` is not given
-        (or None).
+        Default to the current python environment if `pyenv_paths` is empty
+        (the default).
 
         Use importlib_metadata to look up the mapping between packages and their
         provided import names.
         """
-        if pyenv_path is not None:
-            self.pyenv_path = self.determine_package_dir(pyenv_path)
-            if self.pyenv_path is None:
-                raise ValueError(f"Could not find a Python env at {pyenv_path}!")
-        else:
-            self.pyenv_path = None
+        self.package_dirs: Set[Path] = set()  # empty => use sys.path instead
+        if pyenv_paths:
+            package_dirs = {
+                (path, self.determine_package_dir(path)) for path in pyenv_paths
+            }
+            for pyenv_path, package_dir in package_dirs:
+                if package_dir is None:
+                    logger.warning(f"Could not find a Python env at {pyenv_path}!")
+            self.package_dirs = {dir for _, dir in package_dirs if dir is not None}
```

Comment:
I found this code a touch too indirect and hard to follow. Would the following make sense?
```suggestion
      for path in pyenv_paths:
          package_dir = self.determine_package_dir(path)
          if package_dir is None:
              logger.warning(f"Could not find a Python env at {path}!")
          else:
              self.package_dirs.add(package_dir)
```

Reply:
Yes, that reads much better. Thanks! Will adopt, along with a changing the two lines below:
```python
            if not self.package_dirs:
                raise ValueError(f"Could not find any Python env in {pyenv_paths}!")
```
into
```python
        if pyenv_paths and not self.package_dirs:
            raise ValueError(f"Could not find any Python env in {pyenv_paths}!")
```
as an empty `self.package_dirs` is acceptable, but only when `pyenv_paths` was already empty.

Reply:
Fixed in ed8b82a